DateTime.js

```diff
@@ -68,7 +69,11 @@ var Datetime = createClass({
 		if( state.open == undefined )
 			state.open = !this.props.input;
 
+		if (this.props.onOpen) {
+      this.props.onOpen(this.props.defaultValue);
+    }
 		state.currentView = this.props.viewMode;
+    state.initialDate = this.props.defaultValue;
```

Comment:
should not been on state - you can use it directly from props in render function

Reply:
no, since the default value is the actual value of that component.

Reply:
so why should we put this on the state if it doesn't change?

Reply:
it is changed, the defaultValue prop is the actual value. The name is wrong should be called "value"

Reply:
yes but if the component does not change it - we should use it from props

Reply:
We are changing it in each onChange, so when the user change the date this defaultValue prop will no longer be the initial date, it will be the new one.

Reply:
ok got it now - its merely caching the "first" defaultValue 👍